src/theme/JSONSchemaViewer/JSONSchemaElements/SchemaConditional/if-else-then.tsx

```diff
@@ -47,11 +53,38 @@ export default function IfElseThen(props: Props): JSX.Element {
   ) {
     switch (value) {
       case "schema_if":
-        return <CreateNodes schema={schema.if!} />
+        return (
+          <SchemaHierarchyContextProvider
+            value={{
+              level: currentLevel + 1,
+              jsonPointer: `${currentJsonPointer}/if`,
+            }}
+          >
+            <CreateNodes schema={schema.if!} />
+          </SchemaHierarchyContextProvider>
+        )
       case "schema_then":
-        return <CreateNodes schema={schema.then!} />
+        return (
+          <SchemaHierarchyContextProvider
+            value={{
+              level: currentLevel + 1,
+              jsonPointer: `${currentJsonPointer}/then`,
+            }}
+          >
```

Comment:
In my implementation, I avoided the need to `useSchemaHierarchyContext` everywhere by building one of these:

```tsx
export type InnerSchemaHierarchyContextProviderProps = {
  innerJsonPointer: `/${string}`;
  children: JSX.Element;
}

export function InnerSchemaHierarchyContextProvider({
  innerJsonPointer,
  children
}: InnerSchemaHierarchyContextProvider): JSX.Element {
  const { jsonPointer, level } = useSchemaHierarchyContext()

  return (
    <SchemaHierarchyContextProvider value={{
      jsonPointer: `${jsonPointer}${innerJsonPointer}`,
      level: level + 1
    }}>
      {children}
    </SchemaHierarchyContextProvider>
  )
}
```

Thought you might find this nice :)

Reply:
Thanks for the idea. I called it `SchemaHierarchyComponent` and found a missing pointer for [`contentSchema`](https://opis.io/json-schema/2.x/string.html#contentschema) (quite unused in the real world but we never know ...)